lib/stm32wb0/BLE_TransparentMode/Core/Src/stm32wb0x_hal_msp.c

```diff
@@ -0,0 +1,369 @@
+{
+  if(hpka->Instance==PKA)
+  {
+  /* USER CODE BEGIN PKA_MspDeInit 0 */
+
+  /* USER CODE END PKA_MspDeInit 0 */
+    /* Peripheral clock disable */
+    __HAL_RCC_PKA_CLK_DISABLE();
+
+    /* PKA interrupt DeInit */
+    HAL_NVIC_DisableIRQ(PKA_IRQn);
+  /* USER CODE BEGIN PKA_MspDeInit 1 */
+
+  /* USER CODE END PKA_MspDeInit 1 */
+  }
+
+}
+
+/**
+* @brief RADIO MSP Initialization
+* This function configures the hardware resources used in this example
+* @param hradio: RADIO handle pointer
+* @retval None
+*/
+void HAL_RADIO_MspInit(RADIO_HandleTypeDef* hradio)
+{
+  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
+  if(hradio->Instance==RADIO)
+  {
+  /* USER CODE BEGIN RADIO_MspInit 0 */
+
+  /* USER CODE END RADIO_MspInit 0 */
+
+  /** Initializes the peripherals clock
+  */
+    PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_RF;
+    PeriphClkInitStruct.RFClockSelection = RCC_RF_CLK_16M;
+    if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInitStruct) != HAL_OK)
+    {
+      Error_Handler();
+    }
+
+    /* Peripheral clock enable */
+    if (__HAL_RCC_RADIO_IS_CLK_DISABLED())
+    {
+      /* Radio reset */
+      __HAL_RCC_RADIO_FORCE_RESET();
+      __HAL_RCC_RADIO_RELEASE_RESET();
+
+      /* Enable Radio peripheral clock */
+      __HAL_RCC_RADIO_CLK_ENABLE();
+    }
+
+    /* RADIO interrupt Init */
+    HAL_NVIC_SetPriority(RADIO_TXRX_IRQn, 0, 0);
+    HAL_NVIC_EnableIRQ(RADIO_TXRX_IRQn);
+    HAL_NVIC_SetPriority(RADIO_TXRX_SEQ_IRQn, 0, 0);
+    HAL_NVIC_EnableIRQ(RADIO_TXRX_SEQ_IRQn);
+  /* USER CODE BEGIN RADIO_MspInit 1 */
+
+  /* USER CODE END RADIO_MspInit 1 */
+  }
+
+}
+
+/**
+* @brief RADIO MSP De-Initialization
+* This function freeze the hardware resources used in this example
+* @param hradio: RADIO handle pointer
+* @retval None
+*/
+void HAL_RADIO_MspDeInit(RADIO_HandleTypeDef* hradio)
+{
+  if(hradio->Instance==RADIO)
+  {
+  /* USER CODE BEGIN RADIO_MspDeInit 0 */
+
+  /* USER CODE END RADIO_MspDeInit 0 */
+    /* Peripheral clock disable */
+    __HAL_RCC_RADIO_CLK_DISABLE();
+    __HAL_RCC_RADIO_FORCE_RESET();
+    __HAL_RCC_RADIO_RELEASE_RESET();
+
+    /* RADIO interrupt DeInit */
+    HAL_NVIC_DisableIRQ(RADIO_TXRX_IRQn);
+    HAL_NVIC_DisableIRQ(RADIO_TXRX_SEQ_IRQn);
+  /* USER CODE BEGIN RADIO_MspDeInit 1 */
+
+  /* USER CODE END RADIO_MspDeInit 1 */
+  }
+
+}
+
+/**
+* @brief UART MSP Initialization
+* This function configures the hardware resources used in this example
+* @param huart: UART handle pointer
+* @retval None
+*/
+void HAL_UART_MspInit(UART_HandleTypeDef* huart)
+{
+  GPIO_InitTypeDef GPIO_InitStruct = {0};
+  if(huart->Instance==USART1)
+  {
+  /* USER CODE BEGIN USART1_MspInit 0 */
+
+  /* USER CODE END USART1_MspInit 0 */
+    /* Peripheral clock enable */
+    __HAL_RCC_USART1_CLK_ENABLE();
+
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+    __HAL_RCC_GPIOA_CLK_ENABLE();
+    /**USART1 GPIO Configuration
+    PB0     ------> USART1_RX
+    PA1     ------> USART1_TX
+    */
```

Comment:
Is that actually required ?
Don't you rely on Zephyr to enable USART1 ?

Reply:
`HAL_UART_MspInit` and `HAL_UART_MspDeInit `have been excluded from compilation. To minimize the changes I used `#ifndef __ZEPHYR__`